maintenance/health/servicehealthcheck/healthchecker.go

```diff
@@ -0,0 +1,138 @@
+// Copyright © 2019 by PACE Telematics GmbH. All rights reserved.
+// Created at 2019/10/18 Charlotte Pröller
+
+package servicehealthcheck
+
+import (
+	"github.com/gorilla/mux"
+	"github.com/pace/bricks/maintenance/errors"
+	"github.com/pace/bricks/maintenance/log"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+)
+
+type HealthCheck interface {
+	InitHealthCheck() error
+	HealthCheck() (bool, error)
+	CleanUp() error
+}
+
+// ConnectionState can be used for Health Checks
+// It offers a Mutex and the Date of the last check for caching the result
+type ConnectionState struct {
+	LastCheck time.Time
+	isHealthy bool
+	err       error
+	m         sync.Mutex
+}
+
+type handler struct{}
+
+// checks map with all health checks, key: Name of the check
+var checks = make(map[string]HealthCheck)
+var checksLock = sync.RWMutex{}
+
+// initErrors map with all errors that happened in the initialisation of the health checks
+var initErrors = make(map[string]error)
```

Comment:
`initErrors` needs a lock just as `checksLock`

Reply:
initErrors is now sync.Map